### PR TITLE
fix(VField): input autofill style occupies whole container

### LINKED ISSUE #17865: [Bug Report][3.3.9] VTextField autofill background colour not occupying whole text field
### Environment
**Vuetify Version:** 3.3.9
**Vue Version:** 3.3.4
**Browsers:** Chrome 115.0.0.0
**OS:** Windows 10

### Steps to reproduce
1. Open reproduction link
2. Partially type a previous used "username"

### Expected Behavior
Autofill background colour highlight should cover whole v-text-field like input does

### Actual Behavior
Autofill background colour highlight does not cover whole v-text-field

image

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqlUz1v2zAQ/SssF7eAJQ7pUsEJ0qFFp64dogy0eGrZ8gvkSa1h+L/3SCkJ4QSIgUwW3+O9d++OvjvyzyG08wS84zsEG4xEuOkdY7u5kSGUz3IYvEOpHcQVKiDCP2xGDUY9gIwZuQdz3fOvOiZk36WFrudPtCOA2ClBzJ81pRURYy7LVTWDh5CLsl0NzzJq6ZAYP6Gh5lTNygn94CkS4MuOgqI8hilts9FHugpWatPzmy/5t9uJwj3l1i5MeHmiVe3VNK/2y1jCg8nc3kcFsSnHjiVvdJU8xyp9ivOtZWRZ6k5Uy6ZjGqIOyBLgtOxc2+AjsiOLMLITG6O3bEPvZLOMjHRpuTb9ZNf5xvvNNzDGsx8+GvVu8yEbLJIkxrd8UWusDO3v5B09tmNW6Vci9bxjBckYueRzz38hhtQJMShHZQqMnmPrAIULVtzSNREnh9pCo7y9vWqv2o9C6YQ13EKyzT76vzRNEun5trJBPR4usFpvFodPi8MKNfQyUrZ4Ji3Ib6YVRXC0KYiXJjorq1OdUc+SZfdT7040b/ov/Elnox5SGfPdQ5NvilzEstJ9NjxteemTnEsBv/8P0z9nfA==)

### PR DESCRIPTION
**Pull request by yuwu9145**
fixes #17865

image

## Markup:

```vue
<template>
  <v-app>
    <v-container style="width: 850px">
      <h3>Outlined (compact)</h3>
      <v-row>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            variant="outlined"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="outlined"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-combobox
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="outlined"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="outlined"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="outlined"
            density="compact"
            multiple
            chips
          />
        </v-col>
      </v-row>
      <h3>Outlined (comfortable)</h3>
      <v-row>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            variant="outlined"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="outlined"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-combobox
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="outlined"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="outlined"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="outlined"
            density="comfortable"
            multiple
            chips
          />
        </v-col>
      </v-row>
      <h3>Underlined (compact)</h3>
      <v-row>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            variant="underlined"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="underlined"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-combobox
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="underlined"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="underlined"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="underlined"
            density="compact"
            multiple
            chips
          />
        </v-col>
      </v-row>
      <h3>Underlined (comfortable)</h3>
      <v-row>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            variant="underlined"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="underlined"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-combobox
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="underlined"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="underlined"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="underlined"
            density="comfortable"
            multiple
            chips
          />
        </v-col>
      </v-row>
      <h3>Solo (compact)</h3>
      <v-row>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            variant="solo"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="solo"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-combobox
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="solo"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="solo"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="solo"
            density="compact"
            multiple
            chips
          />
        </v-col>
      </v-row>
      <h3>Solo (comfortable)</h3>
      <v-row>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            variant="solo"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="solo"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-combobox
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="solo"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="solo"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            variant="solo"
            density="comfortable"
            multiple
            chips
          />
        </v-col>
      </v-row>
      <h3>Default (compact)</h3>
      <v-row>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-combobox
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            density="compact"
            multiple
            chips
          />
        </v-col>
      </v-row>
      <h3>Default (comfortable)</h3>
      <v-row>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-combobox
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-select
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            density="comfortable"
            multiple
            chips
          />
        </v-col>
      </v-row>

      <v-row>
        <v-col cols="4">
          <v-text-field label="Email:" name="email" type="text" variant="underlined" autocomplete="email" />
        </v-col>
        <v-col cols="4">
          <v-text-field label="Email:" name="email" type="text" variant="underlined" density="comfortable" autocomplete="email" />
        </v-col>
        <v-col cols="4">
          <v-text-field label="Email:" name="email" type="text" variant="underlined" density="compact" autocomplete="email" />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-text-field label="Email:" name="email" type="text" variant="outlined" autocomplete="email" />
        </v-col>
        <v-col cols="4">
          <v-text-field label="Email:" name="email" type="text" variant="outlined" density="comfortable" autocomplete="email" />
        </v-col>
        <v-col cols="4">
          <v-text-field label="Email:" name="email" type="text" variant="outlined" density="compact" autocomplete="email" />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-text-field label="Email:" name="email" type="text" variant="solo" autocomplete="email" />
        </v-col>
        <v-col cols="4">
          <v-text-field label="Email:" name="email" type="text" variant="solo" density="comfortable" autocomplete="email" />
        </v-col>
        <v-col cols="4">
          <v-text-field label="Email:" name="email" type="text" variant="solo" density="compact" autocomplete="email" />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-text-field label="Email:" name="email" type="text" autocomplete="email" />
        </v-col>
        <v-col cols="4">
          <v-text-field label="Email:" name="email" type="text" density="comfortable" autocomplete="email" />
        </v-col>
        <v-col cols="4">
          <v-text-field label="Email:" name="email" type="text" density="compact" autocomplete="email" />
        </v-col>
      </v-row>
      <v-autocomplete
        chips
        multiple
        variant="outlined"
        closableChips
        :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
      />
    </v-container>
    <div style="position: absolute; height: 1px; background-color: green; top: 72px; width: 100%;"></div>
    <div style="position: absolute; height: 1px; background-color: green; top: 170px; width: 100%;"></div>
    <div style="position: absolute; height: 1px; background-color: green; top: 273px; width: 100%;"></div>
    <div style="position: absolute; height: 1px; background-color: green; top: 370px; width: 100%;"></div>
    <div style="position: absolute; height: 1px; background-color: green; top: 457px; width: 100%;"></div>
    <div style="position: absolute; height: 1px; background-color: green; top: 555px; width: 100%;"></div>
    <div style="position: absolute; height: 1px; background-color: green; top: 651px; width: 100%;"></div>
    <div style="position: absolute; height: 1px; background-color: green; top: 749px; width: 100%;"></div>
  </v-app>
</template>

<script setup>
  import { ref } from 'vue'

  const msg = ref('Hello World!')
</script>



```

### CODE REVIEW

**Comment by johnleider** [comment]
Hard to read 
image

**Comment by yuwu9145** [comment]
> Hard to read

Fixed, `v-field__input` should be the lowest z-index inside `v-field__field` container, so its style change won't bother other sibling elements labels

File: packages/vuetify/src/components/VField/VField.sass [hunk 1]
```diff
@@ -131,10 +131,26 @@
   padding-bottom: var(--v-field-input-padding-bottom)
   position: relative
   width: 100%
+  z-index: -1
```
Comment (johnleider) [inline packages/vuetify/src/components/VField/VField.sass:134]: Did you check the docs to make sure this didn't cause any weird side effects?

**Comment by KaelWD** [comment]
This also caused #17991, is it possible to always fill instead of only when `:autofill`?

**Comment by yuwu9145** [comment]
As per my testing:

- Didn't find visual regressions when running docs in my local
- No percy regression: [link](https://percy.io/c458cb73/vuetify/builds/29615908/changed/1642760803?browser=chrome&browser_ids=38&subcategories=unreviewed%2Cchanges_requested&viewLayout=overlay&viewMode=new&width=1280&widths=375%2C1280)
- Checked playground in #17607, no regression being found

Having said above, additional testing by others will be always beneficial and appreciated

File: packages/vuetify/src/components/VField/VField.sass (hunk 1 above)
Comment (yuwu9145) [reply packages/vuetify/src/components/VField/VField.sass]: Did local doc testing after the latest change

**Comment by yuwu9145** [comment]
> This also caused #17991, is it possible to always fill instead of only when `:autofill`?

Yes, made it happen

**Review by johnleider — APPROVED** [review]

**Pull request merged**